docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,434 @@
+3. Additional per-analyzer options can be used to affect which generated logs
+are loaded and what additional processing is performed on them
+
+The analyzer invoked by Brimcap is a wrapper script as referenced in the YAML.
+In addition to taking input on stdin, it also sets Zeek to ignore checksums
+(since these are often set incorrectly on pcaps) and we also disable a couple
+of the excess log outputs. This is very similar to the Zeek Runner script that
+was included with Brim `v0.24.0`.
+
+```
+$ cat zeek-wrapper.sh 
+#!/bin/bash
+zeek -C -r - --exec "event zeek_init() { Log::disable_stream(PacketFilter::LOG); Log::disable_stream(LoadedScripts::LOG); }" local
+```
+
+We use a similar wrapper for Suricata, but here we perform some additional
+steps to work around a known Suricata bug [#4106](https://redmine.openinfosecfoundation.org/issues/4106).
+This bug causes EVE JSON events to be output that contain duplicate keys, which
+Zed is not currently equipped to tolerate ([zed/2523](https://github.com/brimdata/zed/issues/2523)). We therefore pipe the EVE JSON data through [`jq`](https://stedolan.github.io/jq/),
+which has the effect of deduplicating these references. Shell pipeline
+additions such that this illustrate the kinds of processing you may want to
+invoke in your Brimcap customizations, as this may prove useful not just for
+working around bugs but also for providing enhanced functionality.
+
+```
+$ cat suricata-wrapper.sh 
+#!/bin/bash
+suricata -r /dev/stdin
+cat eve.json | jq -c . > deduped-eve.json
+```
+
+Revisiting our Brimcap YAML configuration, we see this output file is
+referenced in a `globs:` setting. This setting can be used to specify one or
+more [glob](https://en.wikipedia.org/wiki/Glob_(programming))-style wildcards
+that isolate a subset of the log outputs from the analyzer that should be
+processed. In this case it was important to only target this "deduped" log
+file, since otherwise the original `eve.json` would still have been processed
+and the error due to the presence of the duplicate fields would have halted the
+`brimcap load`.
+
+```
+analyzers:
+...
+  - cmd: suricata-wrapper.sh
+    globs: ["deduped*.json"]
+```
+
+What follows below the `globs:` setting is a Zed shaper configuration. Whereas
+the Zeek TSV logs contain Zed-compatible rich data types (timestamps, IP
+addresses, etc.), since Suricata's EVE logs are JSON, here we use this shaper
+config to assign better data types as the JSON is being converted for storage
+into the Zed Lake. Out-of-the-box, Brimcap automatically applies this same
+shaping configuration on the EVE output generated from its bundled Suricata.
+Here it's broken out and made part of the configuration YAML such that you can
+further modify it to suit your needs.
+```
+    shaper: |
+      type alert = {
+        timestamp: time,
+        event_type: bstring,
+        src_ip: ip,
+        src_port: port=(uint16),
+        dest_ip: ip,
+        dest_port: port=(uint16),
```

Comment:
Is it necessary to define port again? Could it be `dest_port: port`?

Reply:
Thanks for spotting, @jameskerr. Indeed, I think when Henri whipped up the first shaper configs he was doing it in a programmatic fashion that created these redundant type definitions. I eased them out of the reference Zed shaper in brimdata/zed#2601 but hadn't yet noticed it here. I've pushed a commit to put a single type definition for `port` at the top of the shaper and then reference it throughout.